:sparkles: Add support to a redis based Queue

Andy9822 opened the pull request:
This PR adds support to a Redis-based queue so the rate limit can be used in distributed systems with access to a common Redis server.

The original repo https://github.com/Shopify/limiter implements a simple mechanism to throttle or rate-limit operations in Ruby, developed by Shopify.
Then, I reimplemented https://github.com/nulib/redrate solution (which was basically a Redis-based solution built on top of Shopify's repo).

This Redrate repo was created as a new repo instead of a fork of the original repo. And since it got outdated I opted for sticking to the original Shopify's repo and simply adding the redis-based on the latest version of the original repo.

To test it, I just used the tests for the local queue and adapted them to test the same behaviors in the `DistributedQueue`

Andy9822 committed:
:sparkles: Add support to a redis based Queue

Andy9822 committed:
::bookmark:: Release version 3.0.0

Andy9822 committed:
:white_check_mark: Create tests for DistributedQueue

Andy9822 committed:
:wastebasket::recycle: Refactor use of deprecated methods

danielberkompas reviewed (approved):
I'd need to read more of `limiter`'s source code to fully understand what is going on here, but it seems like a reasonable way to implement a rate limiter with Redis.

Let's see how it works in LS.

Andy9822 committed:
:building_construction::recycle: Use distributed queue as default

Since this project is a fork for internal use at LeadSimple (I just found out that the original repo has no interest in extending it, they want to keep the simple mechanism:
https://github.com/Shopify/limiter/issues/5) I'm gonna adapt it to better fit our needs.

This way, we always will use distributed without having to explicitely say that.
